src/Arcus.Security.Providers.AzureKeyVault/Authentication/CertificateBasedAuthentication.cs

```diff
@@ -37,23 +55,36 @@ public CertificateBasedAuthentication(string clientId, X509Certificate2 certific
         /// <returns>A <see cref="IKeyVaultClient" /> client to use for interaction with the vault</returns>
         public Task<IKeyVaultClient> AuthenticateAsync()
         {
+            _logger.LogTrace("Start authenticating with certificate to the Azure Key Vault...");
```

Comment:
@fgheysels What do you think of this sort logging? Do you want this built-in, opt-in, opt-out?

Reply:
I was just wondering if you cannot 'opt out' or 'opt in' via the regular logging infrastructure ?  
In the configuration file, can't we specify something like this:

```json
{
  "Logging": {
     "LogLevel": {
        "Default": "Warning",
        "Arcus.SecurityProvider": "Trace"
     },
    "ApplicationInsights": {
         "LogLevel": {
            "Default": "Warning",
            "Arcus.SecurityProvider": "Debug"
         }
    }
  }
}
```

Something like that ?  I think this is the standard way of configuring logging for different components.  This is also how you configure logging in Entity Framework for instance.

Reply:
You can definitely do that, the question is more, does this bring any valuable information to the table and if you think Trace is fine. For me Trace is ideal as this brings a lot of noise, but maybe you don't agree.

The verbosity is fully configurable in serilog itself so we don't have to do anything there ourselves, how it works with the config is something I haven't used yet (I don't use app settings)